geoutils/georaster/raster.py

```diff
@@ -229,69 +310,92 @@ def __init__(
         else:
             raise ValueError("filename argument not recognised.")
 
-        self._read_attrs(attrs)
+    @property
+    def nbands(self) -> int:
+        if not self.is_loaded and self._disk_shape is not None:
+            return self._disk_shape[0]
+        return int(self.data.shape[0])
 
-        # Save _masked attribute to be used by self.load()
-        self._masked = masked
+    @property
+    def count(self) -> int:
+        if self._disk_shape is not None:
+            return self._disk_shape[0]
+        return self.nbands
```

Comment:
Wait... here our `nbands` corresponds to rasterio `ds.data.shape[0]`, and then `count` corresponds to `nbands` again? I'm not sure I get the difference! Bit out of touch with those attributes, comments would help!

Reply:
I would very much like input from @adehecq and/or @atedstone here, because I have no idea what the difference is either! I tried to mimic the old behaviour, but I'm not 100% sure it came out right

Reply:
`nbands` is a relic from the old georaster. `count` makes an appearance now because it's an attribute loaded by rasterio. In practical terms it looks like there probably isn't a difference between them. But, there could be some edge cases associated with the user specifying only certain bands to load, in which case `nbands` and `count` might not match. From a quick look, the former appears to be set according to the number of bands loaded, whereas `count` always matches rasterio I think.

Reply:
There is definitely some work needed here to iron out the differences. Perhaps deprecate `nbands` and, if found to be needed by testing, add an equivalent variable named e.g. `count_loaded`. But we should check that such an attribute is actually important to the functionality of the class before re-implementing it!!

Reply:
Yes, the difference between the two is the number of bands in the raster vs the number of bands loaded. I believe count is the former, but I'm not 100% sure without testing it.
I think both are useful in case one loads only a part of the data, and one of the names should be coherent with rasterio's naming convention. But the other one could be renamed if it makes things clearer.

Reply:
What about this:
- `nbands`: Deprecate?
- `count`: The number of bands in the current data (`self.data.shape[0]` if loaded, otherwise `disk_count`)
- `disk_count`: The number of bands in the disk (also the default for `count` if data are not loaded)
- `indexes`: The rio indexes if the Raster was made from a rio dataset, otherwise just 1,2,3 etc.
- `bands`: Deprecate?

Reply:
I think we should stay consistent with rasterio's naming convention and hence have `count` and `indexes` match the definition of rasterio:
- `count`: band count on disk
- `indexes`: The 1-based indexes of each band in the dataset

We can have an equivalent of each for bands loaded, i.e. `count_loaded` and `indexes_loaded` (not sure the latter is really needed...).
What do you think @erikmannerfelt, @atedstone @rhugonnet ? 
Others should be deprecated.

Reply:
Sounds good :+1: